### PR TITLE
docs(python): Alphabetise methods in Python API reference

### PR DESCRIPTION
**Pull request by henryharbeck**
It is evident based on the existing API reference that the methods are intended to be in alphabetical order (and they mostly are).

This does the remainder.

Note that there were a few exceptions where I decided alphabetical order was not better. `strip_chars_end` coming after `strip_chars_start` being one example. Let me know if you have a difference of opinion.

### CODE REVIEW

File: py-polars/docs/source/reference/series/index.rst [hunk 1]
```diff
@@ -19,8 +19,8 @@ This page gives an overview of all public Series methods.
    export
    list
    modify_select
```
Comment (henryharbeck) [inline py-polars/docs/source/reference/series/index.rst:21]: `modify_select` appears in the docs as `Manipulation/selection`, so is correctly before `miscellaneous`

File: py-polars/docs/source/reference/sql/functions/aggregate.rst [hunk 2]
```diff
@@ -202,7 +202,7 @@ Returns the smallest (minimum) of all the elements in the grouping.
 
 STDDEV
 ------
-Returns the standard deviation of all the elements in the grouping.
+Returns the sample standard deviation of all the elements in the grouping.
```
Comment (henryharbeck) [inline py-polars/docs/source/reference/sql/functions/aggregate.rst:205]: drive by

**Comment by mcrumiller** [comment]
Any way to add sorting to `pre-commit`?

**Review by ritchie46 — APPROVED** [review]

**Comment by ritchie46** [comment]
Thanks @henryharbeck

**Comment by henryharbeck** [comment]
@mcrumiller 
> Any way to add sorting to `pre-commit`?

Good idea - will see if I can do something with a small footprint

**Pull request merged**